lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +407,100 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+void ObjectLoadListener::getDebugInfoForObject(
+    const ObjectFile &Obj, const RuntimeDyld::LoadedObjectInfo &L) {
+  OwningBinary<ObjectFile> DebugObjOwner = L.getObjectForDebug(Obj);
+  const ObjectFile &DebugObj = *DebugObjOwner.getBinary();
+
+  // Get the address of the object image for use as a unique identifier
+  const void *ObjData = DebugObj.getData().data();
+  DIContext *DwarfContext = new DWARFContextInMemory(DebugObj);
+
+  // Use symbol info to iterate functions in the object.
+  // TODO: This may have to change when we have funclets
+
+  for (symbol_iterator SI = DebugObj.symbol_begin(), E = DebugObj.symbol_end();
+       SI != E; ++SI) {
+    SymbolRef::Type SymType;
+    if (SI->getType(SymType))
+      continue;
+    if (SymType == SymbolRef::ST_Function) {
+      // Function info
+      StringRef Name;
+      uint64_t Addr;
+      if (SI->getName(Name))
+        continue;
+      if (SI->getAddress(Addr))
+        continue;
+      uint64_t Size = SI->getSize();
+
+      unsigned LastDebugOffset = -1;
+      unsigned NumDebugRanges = 0;
+      ICorDebugInfo::OffsetMapping *OM;
+
+      if (DwarfContext) {
+        DILineInfoTable Lines =
+            DwarfContext->getLineInfoForAddressRange(Addr, Size);
+
+        DILineInfoTable::iterator Begin = Lines.begin();
+        DILineInfoTable::iterator End = Lines.end();
+
+        // Count offset entries. Will skip an entry if the current IL offset
+        // matches
+        // the previous offset.
+        for (DILineInfoTable::iterator It = Begin; It != End; ++It) {
+          int LineNumber = (It->second).Line;
+
+          if (LineNumber != LastDebugOffset) {
+            NumDebugRanges++;
+            LastDebugOffset = LineNumber;
+          }
+        }
+
+        // Reset offset
+        LastDebugOffset = -1;
+        unsigned CurrentDebugEntry = 0;
+
+        if (NumDebugRanges > 0) {
+
+          // Allocate OffsetMapping array
+          unsigned SizeOfArray =
+              (NumDebugRanges) * sizeof(ICorDebugInfo::OffsetMapping);
+          OM = (ICorDebugInfo::OffsetMapping *)Context->JitInfo->allocateArray(
+              SizeOfArray);
+
+          Begin = Lines.begin();
+
+          // Iterate through the debug entries and save IL offset, native
+          // offset, and
+          // source reason
+          for (DILineInfoTable::iterator It = Begin; It != End; ++It) {
+            int Offset = It->first;
+            int LineNumber = (It->second).Line;
+
+            // We store info about if the instruction is being recorded because
+            // it is a call in the column field
+            int IsCall = (It->second).Column;
```

Comment:
I'd change this to `bool IsCall = (It->second).Column == 1;` to avoid the comparison in the ternary on line 491.